src/Test/nlp.jl

```diff
@@ -371,16 +371,108 @@ function test_linear_mixed_complementarity(model::MOI.ModelLike, config::TestCon
         MOI.Complements(4)
     )
     @test MOI.get(model, MOI.TerminationStatus()) == MOI.OPTIMIZE_NOT_CALLED
-    MOI.optimize!(model)
-    @test MOI.get(model, MOI.TerminationStatus()) == config.optimal_status
-    x_val = MOI.get.(model, MOI.VariablePrimal(), x)
-    @test isapprox(
-        x_val, [2.8, 0.0, 0.8, 1.2], atol = config.atol, rtol = config.rtol
+    if config.solve
+        MOI.optimize!(model)
+        @test MOI.get(model, MOI.TerminationStatus()) == config.optimal_status
+        x_val = MOI.get.(model, MOI.VariablePrimal(), x)
+        @test isapprox(
+            x_val, [2.8, 0.0, 0.8, 1.2], atol = config.atol, rtol = config.rtol
+        )
+    end
+end
+
+"""
+    test_qp_mixed_complementarity(model::MOI.ModelLike, config::TestConfig)
+
+Test the solution of the quadratic mixed-complementarity problem:
+
+```
+    min  (x0 - 5)^2 +(2 x1 + 1)^2
+    s.t.  -1.5 x0 + 2 x1 + x2 - 0.5 x3 + x4 = 2
+        x2 complements(3 x0 - x1 - 3)
+        x3 complements(-x0 + 0.5 x1 + 4)
+        x4 complements(-x0 - x1 + 7)
+        x0, x1, x2, x3, x4 >= 0
+
+```
+which rewrites, with auxiliary variables
+
+```
+    min  (x0 - 5)^2 +(2 x1 + 1)^2
+    s.t.  -1.5 x0 + 2 x1 + x2 - 0.5 x3 + x4 = 2  (cf1)
+        3 x0 - x1 - 3 - x5 = 0                 (cf2)
+        -x0 + 0.5 x1 + 4 - x6 = 0              (cf3)
+        -x0 - x1 + 7 - x7 = 0                  (cf4)
+        x2 complements x5
+        x3 complements x6
+        x4 complements x7
+        x0, x1, x2, x3, x4, x5, x6, x7 >= 0
+
+```
+"""
+function test_qp_mixed_complementarity(model::MOI.ModelLike, config::TestConfig)
+    MOI.empty!(model)
+    x = MOI.add_variables(model, 8)
+    MOI.set.(model, MOI.VariablePrimalStart(), x, 0.0)
+    for i in 1:8
+        MOI.add_constraint(model, MOI.SingleVariable(x[i]), MOI.GreaterThan(0.))
```

Comment:
Please add a trailing `0` to `0.`. There are also multiple cases of this further down.